### PR TITLE
Stacked chart option

### PR DESCRIPTION
**Pull request by daniellevinson**
## Description

Add stacked option to line and bar charts.

## Proposed Changes

- Add toggle for stacked if chart has grouping or multiple metrics.

## Screenshots

Screen Shot 2019-06-28 at 12 09 21 PM
Screen Shot 2019-06-28 at 12 09 30 PM
Screen Shot 2019-06-28 at 12 09 41 PM

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/components/navi-visualizations/line-chart.js [hunk 1]
```diff
@@ -199,19 +200,38 @@ export default Component.extend({
...
+      { stacked } = options.style;
+
+    return stacked && seriesData.length ? [Object.keys(seriesData[0]).filter(key => key !== 'x')] : [];
+  }),
+
+  /**
+   * @property {Object} dataConfig - configuration for chart x and y values
+   */
+  dataConfig: computed('c3ChartType', 'seriesData', 'seriesDataGroups', function() {
+    const { c3ChartType, seriesData, seriesDataGroups } = this;
```
Comment (mattgraham1995) [inline packages/core/addon/components/navi-visualizations/line-chart.js]: Can we test to make sure this doesn't break apps on Ember 2.x? I think there's been an issue in the past with this syntax when accessing computed properties

Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/line-chart.js]: Fixed

File: packages/core/addon/components/navi-visualizations/line-chart.js [hunk 2]
```diff
@@ -199,19 +200,40 @@ export default Component.extend({
...
+      { stacked } = options.style;
+
+    return stacked && seriesData.length ? [Object.keys(seriesData[0]).filter(key => key !== 'x')] : [];
+  }),
+
+  /**
+   * @property {Object} dataConfig - configuration for chart x and y values
+   */
+  dataConfig: computed('c3ChartType', 'seriesData', 'seriesDataGroups', function() {
+    const c3ChartType = get(this, 'c3ChartType'),
```
Comment (jkusa) [inline packages/core/addon/components/navi-visualizations/line-chart.js]: Can we do `this.{prop}`?

File: packages/core/addon/components/navi-visualizations/line-chart.js [hunk 3]
```diff
@@ -199,19 +200,40 @@ export default Component.extend({
...
+
+  /**
+   * @property {Array} seriesDataGroups - chart series groups for stacking
+   */
+  seriesDataGroups: computed('options', 'seriesData', function() {
+    const seriesData = get(this, 'seriesData'),
+      options = merge({}, DEFAULT_OPTIONS, this.options),
+      { stacked } = options.style;
+
+    return stacked && seriesData.length ? [Object.keys(seriesData[0]).filter(key => key !== 'x')] : [];
```
Comment (jkusa) [inline packages/core/addon/components/navi-visualizations/line-chart.js]: Can we comment this?

File: packages/core/addon/components/navi-visualizations/line-chart.js (hunk 2 above)
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/line-chart.js]: `this.seriesData` breaks in Ember 2.x

File: packages/core/addon/components/navi-visualizations/line-chart.js (hunk 3 above)
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/line-chart.js]: done

File: packages/core/addon/templates/components/visualization-config/line-chart.hbs [hunk 4]
```diff
@@ -1,12 +1,23 @@
 {{!-- Copyright 2019, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
 <span class="line-chart-config__header">Chart Style Options</span>
 <div class="line-chart-config__style-options">
+  {{#if displayStackOption}}
+    <div class="line-chart-config__stacked-opt">
```
Comment (jkusa) [inline packages/core/addon/templates/components/visualization-config/line-chart.hbs]: Can we move this toggle under the select with the other toggle?

File: packages/core/tests/integration/components/visualization-config/line-chart-test.js [hunk 5]
```diff
@@ -67,4 +87,27 @@ module('Integration | Component | visualization config/line chart', function(hoo
 
     await run(() => click('.mock'));
   });
+
+  test('displayStackOption', async function(assert) {
+    assert.expect(2);
+
+    await render(Template);
+
+    assert.ok(
```
Comment (jkusa) [inline packages/core/tests/integration/components/visualization-config/line-chart-test.js]: Can we use assert.dom like we do here: https://github.com/yahoo/navi/blob/f99912bd4facb9f7183ca570eec248589705cb59/packages/dashboards/tests/integration/components/dashboard-filters-expanded-test.js#L60

File: packages/core/addon/components/navi-visualizations/line-chart.js (hunk 2 above)
Comment (jkusa) [reply packages/core/addon/components/navi-visualizations/line-chart.js]: Hmm, did you test this?

File: packages/core/addon/components/visualization-config/line-chart.js [hunk 6]
```diff
@@ -33,6 +34,16 @@ export default Component.extend({
    */
   typePrefix: 'visualization-config/chart-type/',
 
+  /**
+   * @property {Boolean} displayStackOption - whether to display the `stacked` toggle
+   */
+  displayStackOption: computed('type', 'response.rows', function() {
```
Comment (jkusa) [inline packages/core/addon/components/visualization-config/line-chart.js]: are you using `response.rows`?

Comment (jkusa) [reply packages/core/addon/components/visualization-config/line-chart.js]: Should this be `request`?

File: packages/core/addon/components/navi-visualizations/line-chart.js (hunk 2 above)
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/line-chart.js]: yes, but I'll take another look

File: packages/core/addon/templates/components/visualization-config/line-chart.hbs (hunk 4 above)
Comment (daniellevinson) [reply packages/core/addon/templates/components/visualization-config/line-chart.hbs]: done
Screen Shot 2019-07-03 at 11 47 05 AM

File: packages/core/addon/components/visualization-config/line-chart.js (hunk 6 above)
Comment (daniellevinson) [reply packages/core/addon/components/visualization-config/line-chart.js]: yes, fixed

File: packages/core/addon/components/visualization-config/line-chart.js [hunk 7]
```diff
@@ -33,6 +35,20 @@ export default Component.extend({
    */
   typePrefix: 'visualization-config/chart-type/',
 
+  /**
+   * @property {Boolean} displayStackOption - whether to display the `stacked` toggle
+   */
+  displayStackOption: computed('type', 'request', function() {
+    if (!featureFlag('enableChartStacking')) {
```
Comment (daniellevinson) [inline packages/core/addon/components/visualization-config/line-chart.js]: feature flag added

File: packages/core/addon/components/navi-visualizations/line-chart.js (hunk 2 above)
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/line-chart.js]: @jkusa you mean in #421?
Now that I merged it in I see that this, too, breaks in 2.x.
Looks like `this.{prop}` when using with cp's returns a `ComputedProperty` and not the value.
@kevinhinterlong have you tested it with 2.x? I might be missing something here.

File: packages/core/addon/components/navi-visualizations/line-chart.js [hunk 8]
```diff
@@ -201,19 +202,56 @@ export default Component.extend({
   }),
 
   /**
-   * @property {Object} data - configuration for chart x and y values
+   * @property {Array} seriesData - chart series data
    */
-  dataConfig: computed('firstModel', 'seriesConfig', 'c3ChartType', function() {
+  seriesData: computed('firstModel', 'seriesConfig', function() {
```
Comment (jkusa) [inline packages/core/addon/components/navi-visualizations/line-chart.js]: Can we add `builder` as a cache key?

**Pull request merged**